src/main/java/org/kitteh/irc/client/library/feature/InteractionDefaults.java

```diff
@@ -0,0 +1,164 @@
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.feature;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InteractionDefaults implements InteractionMap {
+    protected final Map<Interaction, String> defaults = new HashMap<>();
+    protected final String defaultString;
+
+    /**
+     * Creates a InteractionDefaults object with a custom string to default
+     * to when an Interaction key is not set.
+     *
+     * Will throw an IllegalArgumentException is you set the
+     * defaultString value to "". Use null instead.
+     *
+     * @param defaultString a default string to use for when a key is not set
+     */
+    public InteractionDefaults(@Nullable String defaultString) {
+        if (defaultString != null && defaultString.isEmpty()) {
+            // Both "" and null are legal here, technically. But for KICL to perform best (and to avoid != null && string.isEmpty() checks),
+            // we enforce a requirement on null.
+            throw new IllegalArgumentException("defaultString may not be an empty string, use null instead.");
+        }
+        this.defaultString = defaultString;
+    }
+
+    /**
+     * Creates a InteractionDefaults object with predefined string to default
+     * to when an Interaction key is not set.
+     */
+    public InteractionDefaults() {
+        this.defaultString = null;
+    }
+
+    /**
+     * Sets a default message for the Interaction.
+     *
+     * Will throw an IllegalArgumentException is you set the
+     * defaultString value to "". Use null instead.
+     *
+     * Will throw a RuntimeException if you try to define a value
+     * that is already defined. This often indicates a problem and
+     * #replaceDefault should be used instead.
+     *
+     * @param key The interaction to set a value for
+     * @param defaultString The default value to assume when nothing is provided for this Interaction
+     * @return Returns this to permit chaining
+     */
+    public InteractionDefaults setDefault(Interaction key, @Nullable String defaultString) {
+        if (defaultString != null && defaultString.isEmpty()) {
```

Comment:
Since you have the exact same snippet of code here and in the constructor, you can move it to a method and call said method to reduce duplication.